crates/sui-core/src/authority_aggregator.rs

```diff
@@ -961,46 +971,32 @@ where
                         async move { client.handle_transaction(transaction_ref.clone()).await },
                     )
                 },
-                |mut state, name, weight, result| {
+                |mut state, name, weight, response| {
                     Box::pin(async move {
-                        match result {
-                            Ok(response) => match response {
-                                VerifiedTransactionInfoResponse::Signed(signed) => {
-                                    if let Some(cert) = self
-                                        .handle_response_with_signed_transaction(
-                                            &mut state, signed, name, weight,
-                                        )
-                                    {
-                                        return ReduceOutput::Success(cert);
-                                    }
-                                }
-                                VerifiedTransactionInfoResponse::Executed(cert, effects) => {
-                                    if let Some(result) = self
-                                        .handle_response_with_certified_transaction(
-                                            &mut state, cert, effects, name, weight,
-                                        )
-                                    {
-                                        return ReduceOutput::Success(result);
-                                    }
-                                }
-                            },
-                            Err(err) => {
-                                self.handle_response_with_err(
-                                    &mut state, name, weight, tx_digest, err,
-                                );
+                        match self.handle_process_transaction_response(
+                            tx_digest, &mut state, response, name, weight,
+                        ) {
+                            Ok(Some(result)) => {
+                                self.record_process_transaction_metrics(tx_digest, &state);
```

Comment:
move this and line 1010 to 1006?

Reply:
Right now we don't return state on success. So I had to record the metrics here.